frontend/src/modules/history/components/Translation.js

```diff
@@ -99,8 +118,50 @@ export default class Translation extends React.Component<Props> {
         </a>
     }
 
+    toggleDiff = () => {
+        this.setState((state) => {
+            return { isDiffVisible: !state.isDiffVisible };
+        });
+    }
+
+    getDiff(base: string, target: string) {
+        const diff = dmp.diff_main(base, target);
+
+        dmp.diff_cleanupSemantic(diff);
+        dmp.diff_cleanupEfficiency(diff);
+
+        return diff;
+    }
+
+    renderDiff(translation: DBTranslation, activeTranslation: DBTranslation) {
```

Comment:
Nit: I would directly pass the strings here, instead of the full objects, because you don't need the objects anyway and that makes the function more "dumb" and thus easier to test.